internal/ethapi/api.go

```diff
@@ -885,6 +885,8 @@ func (diff *StateOverride) Apply(state *state.StateDB) error {
 			}
 		}
 	}
+	// Now commit the changes
+	state.Commit(false)
```

Comment:
That will actually write into the trie database, which is a big nono. It might still be "just in ram", but it will compete with the memory cache of real trie nodes. I'd wager it also creates a diff layer on top of the snapshots if one exists.

Reply:
Gotcha! Using finalize seems to work  too